PiG: error with PigContext

When I ran bacon - pig with this options (correct at the time, ignore the bacon.sh its just a wrapper for the java run of the bacon jar), the ALTERED maven repo zip is because there is lose files (README) in the top level that won't work when used in this situation
```
bacon.sh pig run -t --skipRepo --skipPncUpdate --skipBuilds --skipSources --skipSharedContent -r ../product-files-generator/target/vertx-3.9.1.CR1/rhb-vertx-3.9.1.CR1-maven-repository-ALTERED.zip ../build-configurations/Vert.X/3.9/
```
I get this error

```
[ERROR] - Something wrong happened: 
java.lang.RuntimeException: failed to store PigContext
	at org.jboss.pnc.bacon.pig.impl.PigContext.storeContext(PigContext.java:112)
	at org.jboss.pnc.bacon.pig.PigFacade.run(PigFacade.java:119)
	at org.jboss.pnc.bacon.pig.Pig$Run.doExecute(Pig.java:212)
	at org.jboss.pnc.bacon.pig.Pig$Run.doExecute(Pig.java:104)
	at org.jboss.pnc.bacon.pig.Pig$PigCommand.lambda$execute$0(Pig.java:96)
	at org.jboss.pnc.bacon.common.cli.AbstractCommand.executeHelper(AbstractCommand.java:205)
	at org.jboss.pnc.bacon.pig.Pig$PigCommand.execute(Pig.java:94)
	at org.aesh.command.impl.operator.EndOperator.execute(EndOperator.java:41)
	at org.aesh.command.impl.Executions$ExecutionImpl.execute(Executions.java:213)
	at org.aesh.command.impl.AeshCommandRuntime.executeCommand(AeshCommandRuntime.java:161)
	at org.jboss.pnc.bacon.cli.App.run(App.java:97)
	at org.jboss.pnc.bacon.cli.App.main(App.java:128)
Caused by: com.fasterxml.jackson.databind.JsonMappingException: (was java.lang.NullPointerException) (through reference chain: org.jboss.pnc.bacon.pig.impl.PigContext["pigConfiguration"]->org.jboss.pnc.bacon.pig.impl.config.PigConfiguration["defaultBuildParameters"]->org.jboss.pnc.bacon.pig.impl.config.BuildConfig["shortScmURIPath"])
	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:394)
	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:353)
	at com.fasterxml.jackson.databind.ser.std.StdSerializer.wrapAndThrow(StdSerializer.java:316)
	at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:730)
	at com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:166)
	at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:727)
	at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:722)
	at com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:166)
	at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:727)
	at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:722)
	at com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:166)
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider._serialize(DefaultSerializerProvider.java:480)
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:400)
	at com.fasterxml.jackson.databind.ObjectWriter$Prefetch.serialize(ObjectWriter.java:1429)
	at com.fasterxml.jackson.databind.ObjectWriter._configAndWriteValue(ObjectWriter.java:1135)
	at com.fasterxml.jackson.databind.ObjectWriter.writeValue(ObjectWriter.java:960)
	at org.jboss.pnc.bacon.pig.impl.PigContext.storeContext(PigContext.java:110)
	... 11 common frames omitted
Caused by: java.lang.NullPointerException: null
	at java.net.URI$Parser.parse(URI.java:3042)
	at java.net.URI.<init>(URI.java:588)
	at org.jboss.pnc.bacon.pig.impl.config.BuildConfig.getShortScmURIPath(BuildConfig.java:208)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:688)
	at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:722)
	... 24 common frames omitted
```

--- a/pig/src/main/java/org/jboss/pnc/bacon/pig/impl/config/BuildConfig.java
+++ b/pig/src/main/java/org/jboss/pnc/bacon/pig/impl/config/BuildConfig.java
@@ -202,11 +202,23 @@ public class BuildConfig {
     }
 
     // Work around for NCL-3670
+
+    /**
+     * Return the short scm uri path of the scm urls (scmUrl or externalScmUrl respectively)
+     *
+     * If the scmUrl or the getExternalScmUrl is null, this method will also return null
+     *
+     * @return short scm uri path
+     */
     public String getShortScmURIPath() {
         String scmUrl = getScmUrl() != null ? getScmUrl() : getExternalScmUrl();
         try {
-            URI scmUri = new URI(scmUrl);
-            return scmUri.getPath();
+            if (scmUrl != null) {
+                URI scmUri = new URI(scmUrl);
+                return scmUri.getPath();
+            } else {
+                return null;
+            }
         } catch (URISyntaxException e) {
             throw new RuntimeException("Invalid scm URI: " + getScmUrl(), e);
         }

